src/NServiceBus.Core/Sagas/SimpleSaga.cs

```diff
@@ -0,0 +1,85 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Linq.Expressions;
+    using System.Reflection;
+
+    /// <summary>
+    /// This class is used to define sagas containing data and handling a message.
+    /// To handle more message types, implement <see cref="IHandleMessages{T}" />
+    /// for the relevant types.
+    /// To signify that the receipt of a message should start this saga,
+    /// implement <see cref="IAmStartedByMessages{T}" /> for the relevant message type.
+    /// </summary>
+    /// <typeparam name="TSagaData">A type that implements <see cref="IContainSagaData" />.</typeparam>
+    public abstract class SimpleSaga<TSagaData> : Saga where TSagaData : IContainSagaData, new()
+    {
+
+        /// <summary>
+        /// Gets the name of the correlation property for <typeparamref name="TSagaData"/>.
+        /// </summary>
+        protected abstract string CorrelationPropertyName { get; }
```

Comment:
@timbussmann i did consider that. It was actually my first version. u can see i removed it here https://github.com/Particular/NServiceBus/pull/4546/commits/d6eee1d5c4c4bd443ca97d315eac357286632847

Comparing the two

 * both are strong typed. ie nameof is just typed as an expression
 * both are prone to mis-use. eg pass a bad string  or pass a bad expression: `CorrelationProperty = sagaData => this.Data.CorrelationId`
 * both would be validated at compile time in the sql persistence. ie ensure the returned value maps to a property on the saga data
 * both are as un-intuitive/un-dicoverable as each other. ie it think most people using the api for the first time would need to look at the doco.

The reason i went with a string is it is vastly easier to reason about at compile time.

eg from the IL perspective we would most likely interrogate both of the following as valid

    CorrelationProperty = sagaData => this.Data.MyCorrelationId

and

   CorrelationProperty = sagaData => sagaData.MyCorrelationId

Where it is clear only the second one is. The first would probably require a secondary runtime check the first time the saga is executed.